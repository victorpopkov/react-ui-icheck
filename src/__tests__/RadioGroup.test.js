import React from 'react';
import expect from 'expect';
import { mount } from 'enzyme';
import { Radio, RadioGroup } from '../index'; // eslint-disable-line sort-imports

describe('<RadioGroup /> component', () => {
  const testRadioGroup = (
    <RadioGroup name="test">
      <Radio />
    </RadioGroup>
  );

  it('should have the default props values when mounted', () => {
    const wrapper = mount(testRadioGroup);
    expect(wrapper.props()).toMatchSnapshot();
  });

  describe('should render the corresponding HTML', () => {
    it('when only the default props are set', () => {
      const wrapper = mount(testRadioGroup);
      expect(wrapper).toMatchSnapshot();
    });

    it('when the radioWrapClassName and radioWrapTag props values are set', () => {
      const wrapper = mount(
        <RadioGroup name="test" radioWrapClassName="test" radioWrapTag="div">
          <Radio />
        </RadioGroup>
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('should update the state', () => {
    it('when the value prop has changed', () => {
      const spy = jest.spyOn(Radio.prototype, 'componentWillReceiveProps');
      const wrapper = mount(testRadioGroup);

      expect(spy).toHaveBeenCalledTimes(0);
      expect(wrapper.props().value).toBe(null);
      expect(wrapper.state().value).toBe('');

      wrapper.setProps({ value: 'test' });

      expect(spy).toHaveBeenCalledTimes(1);
      expect(wrapper.props().value).toBe('test');
      expect(wrapper.state().value).toBe('test');
    });

    describe('when handling the <input />', () => {
      it('onChange()', () => {
        const spy = jest.fn();
        const wrapper = mount(
          <RadioGroup
            name="test"
            radioWrapClassName="test"
            radioWrapTag="div"
            onChange={spy}
          >
            <Radio value="1" />
            <Radio value="2" />
          </RadioGroup>
        );

        expect(wrapper).toMatchSnapshot();

        expect(spy).toHaveBeenCalledTimes(0);
        expect(wrapper.state().value).toBe('');

        wrapper.find('input').first().simulate('change');

        expect(spy).toHaveBeenCalledTimes(1);
        expect(wrapper.state().value).toBe('1');

        wrapper.find('input').last().simulate('change');

        expect(spy).toHaveBeenCalledTimes(2);
        expect(wrapper.state().value).toBe('2');
      });
    });
  });
});
