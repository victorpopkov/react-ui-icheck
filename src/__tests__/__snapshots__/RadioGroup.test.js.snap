// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> component should have the default props values when mounted 1`] = `
Object {
  "children": <Radio
    activeClass="active"
    activeClassName="active"
    aria={false}
    checked={false}
    checkedClass="checked"
    checkedClassName="checked"
    className={null}
    defaultChecked={false}
    determinateClass={null}
    disabled={false}
    disabledClass="disabled"
    disabledClassName="disabled"
    focusClass="focus"
    hoverClass="hover"
    id={null}
    increaseArea=""
    indeterminate={true}
    indeterminateClass="indeterminate"
    inheritClass={false}
    inheritID={false}
    insert={null}
    isMobile={false}
    label={null}
    labelClassName={null}
    labelHover={true}
    labelHoverClass="hover"
    labelTag="span"
    labelTagClassName={null}
    name={null}
    onBlur={null}
    onChange={null}
    onFocus={null}
    onMouseDown={null}
    onMouseLeave={null}
    onMouseUp={null}
    onTouchEnd={null}
    onTouchStart={null}
    radioClass="iradio"
    type={null}
    value=""
  />,
  "className": null,
  "defaultValue": null,
  "name": "test",
  "onChange": null,
  "radioWrapClassName": null,
  "radioWrapTag": null,
  "value": null,
}
`;

exports[`<RadioGroup /> component should render the corresponding HTML when only the default props are set 1`] = `
<RadioGroup
  className={null}
  defaultValue={null}
  name="test"
  onChange={null}
  radioWrapClassName={null}
  radioWrapTag={null}
  value={null}
>
  <div
    className={null}
  >
    <Radio
      activeClass="active"
      activeClassName="active"
      aria={false}
      checked={true}
      checkedClass="checked"
      checkedClassName="checked"
      className={null}
      defaultChecked={false}
      determinateClass={null}
      disabled={false}
      disabledClass="disabled"
      disabledClassName="disabled"
      focusClass="focus"
      hoverClass="hover"
      id={null}
      increaseArea=""
      indeterminate={true}
      indeterminateClass="indeterminate"
      inheritClass={false}
      inheritID={false}
      insert={null}
      isMobile={false}
      key=".0"
      label={null}
      labelClassName={null}
      labelHover={true}
      labelHoverClass="hover"
      labelTag="span"
      labelTagClassName={null}
      name={null}
      onBlur={null}
      onChange={[Function]}
      onFocus={null}
      onMouseDown={null}
      onMouseLeave={null}
      onMouseUp={null}
      onTouchEnd={null}
      onTouchStart={null}
      radioClass="iradio"
      type={null}
      value=""
    >
      <Input
        activeClass="active"
        activeClassName="active"
        aria={false}
        checkboxClass="icheckbox"
        checkboxClassName="icheckbox"
        checked={true}
        checkedClass="checked"
        checkedClassName="checked"
        className={null}
        defaultChecked={false}
        disabled={false}
        disabledClass="disabled"
        disabledClassName="disabled"
        focusClass="focus"
        focusClassName="focus"
        hoverClass="hover"
        hovered={false}
        id={null}
        increaseArea=""
        indeterminate={true}
        indeterminateClass="indeterminate"
        inheritClass={false}
        inheritID={false}
        inputType="radio"
        insert={null}
        isMobile={false}
        label={null}
        name={null}
        onBlur={null}
        onChange={[Function]}
        onFocus={null}
        radioClass="iradio"
        value=""
      >
        <div
          className="iradio indeterminate checked"
        >
          <input
            checked={true}
            disabled={false}
            label={null}
            name={null}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "opacity": 0,
                "position": "absolute",
              }
            }
            type="radio"
            value=""
          />
          <ins
            className="iCheck-helper"
            onClick={[Function]}
            role="presentation"
            style={
              Object {
                "background": "#fff",
                "border": 0,
                "cursor": "pointer",
                "display": "block",
                "height": "100%",
                "left": "0%",
                "margin": 0,
                "opacity": 0,
                "padding": 0,
                "position": "absolute",
                "top": "0%",
                "width": "100%",
              }
            }
          />
        </div>
      </Input>
    </Radio>
  </div>
</RadioGroup>
`;

exports[`<RadioGroup /> component should render the corresponding HTML when the radioWrapClassName and radioWrapTag props values are set 1`] = `
<RadioGroup
  className={null}
  defaultValue={null}
  name="test"
  onChange={null}
  radioWrapClassName="test"
  radioWrapTag="div"
  value={null}
>
  <div
    className={null}
  >
    <div
      className="test"
      key=".0"
    >
      <Radio
        activeClass="active"
        activeClassName="active"
        aria={false}
        checked={true}
        checkedClass="checked"
        checkedClassName="checked"
        className={null}
        defaultChecked={false}
        determinateClass={null}
        disabled={false}
        disabledClass="disabled"
        disabledClassName="disabled"
        focusClass="focus"
        hoverClass="hover"
        id={null}
        increaseArea=""
        indeterminate={true}
        indeterminateClass="indeterminate"
        inheritClass={false}
        inheritID={false}
        insert={null}
        isMobile={false}
        key=""
        label={null}
        labelClassName={null}
        labelHover={true}
        labelHoverClass="hover"
        labelTag="span"
        labelTagClassName={null}
        name={null}
        onBlur={null}
        onChange={[Function]}
        onFocus={null}
        onMouseDown={null}
        onMouseLeave={null}
        onMouseUp={null}
        onTouchEnd={null}
        onTouchStart={null}
        radioClass="iradio"
        type={null}
        value=""
      >
        <Input
          activeClass="active"
          activeClassName="active"
          aria={false}
          checkboxClass="icheckbox"
          checkboxClassName="icheckbox"
          checked={true}
          checkedClass="checked"
          checkedClassName="checked"
          className={null}
          defaultChecked={false}
          disabled={false}
          disabledClass="disabled"
          disabledClassName="disabled"
          focusClass="focus"
          focusClassName="focus"
          hoverClass="hover"
          hovered={false}
          id={null}
          increaseArea=""
          indeterminate={true}
          indeterminateClass="indeterminate"
          inheritClass={false}
          inheritID={false}
          inputType="radio"
          insert={null}
          isMobile={false}
          label={null}
          name={null}
          onBlur={null}
          onChange={[Function]}
          onFocus={null}
          radioClass="iradio"
          value=""
        >
          <div
            className="iradio indeterminate checked"
          >
            <input
              checked={true}
              disabled={false}
              label={null}
              name={null}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "opacity": 0,
                  "position": "absolute",
                }
              }
              type="radio"
              value=""
            />
            <ins
              className="iCheck-helper"
              onClick={[Function]}
              role="presentation"
              style={
                Object {
                  "background": "#fff",
                  "border": 0,
                  "cursor": "pointer",
                  "display": "block",
                  "height": "100%",
                  "left": "0%",
                  "margin": 0,
                  "opacity": 0,
                  "padding": 0,
                  "position": "absolute",
                  "top": "0%",
                  "width": "100%",
                }
              }
            />
          </div>
        </Input>
      </Radio>
    </div>
  </div>
</RadioGroup>
`;

exports[`<RadioGroup /> component should update the state when handling the <input /> onChange() 1`] = `
<RadioGroup
  className={null}
  defaultValue={null}
  name="test"
  onChange={[MockFunction]}
  radioWrapClassName="test"
  radioWrapTag="div"
  value={null}
>
  <div
    className={null}
  >
    <div
      className="test"
      key=".0"
    >
      <Radio
        activeClass="active"
        activeClassName="active"
        aria={false}
        checked={false}
        checkedClass="checked"
        checkedClassName="checked"
        className={null}
        defaultChecked={false}
        determinateClass={null}
        disabled={false}
        disabledClass="disabled"
        disabledClassName="disabled"
        focusClass="focus"
        hoverClass="hover"
        id={null}
        increaseArea=""
        indeterminate={true}
        indeterminateClass="indeterminate"
        inheritClass={false}
        inheritID={false}
        insert={null}
        isMobile={false}
        key="1"
        label={null}
        labelClassName={null}
        labelHover={true}
        labelHoverClass="hover"
        labelTag="span"
        labelTagClassName={null}
        name={null}
        onBlur={null}
        onChange={[Function]}
        onFocus={null}
        onMouseDown={null}
        onMouseLeave={null}
        onMouseUp={null}
        onTouchEnd={null}
        onTouchStart={null}
        radioClass="iradio"
        type={null}
        value="1"
      >
        <Input
          activeClass="active"
          activeClassName="active"
          aria={false}
          checkboxClass="icheckbox"
          checkboxClassName="icheckbox"
          checked={false}
          checkedClass="checked"
          checkedClassName="checked"
          className={null}
          defaultChecked={false}
          disabled={false}
          disabledClass="disabled"
          disabledClassName="disabled"
          focusClass="focus"
          focusClassName="focus"
          hoverClass="hover"
          hovered={false}
          id={null}
          increaseArea=""
          indeterminate={true}
          indeterminateClass="indeterminate"
          inheritClass={false}
          inheritID={false}
          inputType="radio"
          insert={null}
          isMobile={false}
          label={null}
          name={null}
          onBlur={null}
          onChange={[Function]}
          onFocus={null}
          radioClass="iradio"
          value="1"
        >
          <div
            className="iradio indeterminate"
          >
            <input
              checked={false}
              disabled={false}
              label={null}
              name={null}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "opacity": 0,
                  "position": "absolute",
                }
              }
              type="radio"
              value="1"
            />
            <ins
              className="iCheck-helper"
              onClick={[Function]}
              role="presentation"
              style={
                Object {
                  "background": "#fff",
                  "border": 0,
                  "cursor": "pointer",
                  "display": "block",
                  "height": "100%",
                  "left": "0%",
                  "margin": 0,
                  "opacity": 0,
                  "padding": 0,
                  "position": "absolute",
                  "top": "0%",
                  "width": "100%",
                }
              }
            />
          </div>
        </Input>
      </Radio>
    </div>
    <div
      className="test"
      key=".1"
    >
      <Radio
        activeClass="active"
        activeClassName="active"
        aria={false}
        checked={false}
        checkedClass="checked"
        checkedClassName="checked"
        className={null}
        defaultChecked={false}
        determinateClass={null}
        disabled={false}
        disabledClass="disabled"
        disabledClassName="disabled"
        focusClass="focus"
        hoverClass="hover"
        id={null}
        increaseArea=""
        indeterminate={true}
        indeterminateClass="indeterminate"
        inheritClass={false}
        inheritID={false}
        insert={null}
        isMobile={false}
        key="2"
        label={null}
        labelClassName={null}
        labelHover={true}
        labelHoverClass="hover"
        labelTag="span"
        labelTagClassName={null}
        name={null}
        onBlur={null}
        onChange={[Function]}
        onFocus={null}
        onMouseDown={null}
        onMouseLeave={null}
        onMouseUp={null}
        onTouchEnd={null}
        onTouchStart={null}
        radioClass="iradio"
        type={null}
        value="2"
      >
        <Input
          activeClass="active"
          activeClassName="active"
          aria={false}
          checkboxClass="icheckbox"
          checkboxClassName="icheckbox"
          checked={false}
          checkedClass="checked"
          checkedClassName="checked"
          className={null}
          defaultChecked={false}
          disabled={false}
          disabledClass="disabled"
          disabledClassName="disabled"
          focusClass="focus"
          focusClassName="focus"
          hoverClass="hover"
          hovered={false}
          id={null}
          increaseArea=""
          indeterminate={true}
          indeterminateClass="indeterminate"
          inheritClass={false}
          inheritID={false}
          inputType="radio"
          insert={null}
          isMobile={false}
          label={null}
          name={null}
          onBlur={null}
          onChange={[Function]}
          onFocus={null}
          radioClass="iradio"
          value="2"
        >
          <div
            className="iradio indeterminate"
          >
            <input
              checked={false}
              disabled={false}
              label={null}
              name={null}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "opacity": 0,
                  "position": "absolute",
                }
              }
              type="radio"
              value="2"
            />
            <ins
              className="iCheck-helper"
              onClick={[Function]}
              role="presentation"
              style={
                Object {
                  "background": "#fff",
                  "border": 0,
                  "cursor": "pointer",
                  "display": "block",
                  "height": "100%",
                  "left": "0%",
                  "margin": 0,
                  "opacity": 0,
                  "padding": 0,
                  "position": "absolute",
                  "top": "0%",
                  "width": "100%",
                }
              }
            />
          </div>
        </Input>
      </Radio>
    </div>
  </div>
</RadioGroup>
`;
